GVFS/GVFS/CommandLine/SparseVerb.cs

```diff
@@ -93,6 +93,44 @@ private enum SetDirectoryTimeResult
 
         protected override string VerbName => SparseVerbName;
 
+        internal static bool PathCoveredBySparseFolders(ref int index, string statusOutput, HashSet<string> sparseFolders, out string gitPath)
+        {
+            int endOfPathIndex = statusOutput.IndexOf(StatusPathSeparatorToken, index);
+            gitPath = statusOutput.Substring(index, endOfPathIndex - index);
+            index = endOfPathIndex + 1;
+
+            string filePath = gitPath.Replace(GVFSConstants.GitPathSeparator, Path.DirectorySeparatorChar);
+            if (sparseFolders.Any(x => filePath.StartsWith(x + Path.DirectorySeparatorChar, GVFSPlatform.Instance.Constants.PathComparison)))

```

Comment:
Since `sparseFolders` is a `HashSet<string>` you are probably better off stripping parents from `filePath` and checking `sparseFolders.Contains(filePath)`. [Here is similar code in the `git sparse-checkout` implementation](https://github.com/derrickstolee/git/blob/7577ffc034be90bd60da17e8f3189d999e7f8149/dir.c#L731-L756).

Reply:
Filed follow-up #1608 to address this.